TopMovies/Core/Reducer/MainReducer.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  MainReducer.swift
+//  TopMovies
+//
+//  Created by anikolaenko on 02.02.2021.
+//
+
+import ReSwift
+
+func mainReducer(action: Action, state: MainState?) -> MainState {
+    let state = state ?? MainState(appFlowState: .launching,
+                                   moviesListState: MoviesListState(moviesList: .initial))
+    return MainState(appFlowState: AppFlowState.reduce(action: action, state: state.appFlowState),
+                     moviesListState: MoviesListState.reduce(action: action, state: state.moviesListState))
+}
+
+extension AppFlowState {
+    static func reduce(action: Action, state: AppFlowState) -> AppFlowState {
+        switch action {
+        case AppFlowAction.applicationDidFinishLaunching, AppFlowAction.applicationWillEnterForeground:
+            return .foreground
+        case AppFlowAction.applicationDidEnterBackground:
+            return .background
+        case AppFlowAction.applicationWillTerminate:
+            return.terminating
+        default: return state
+        }
+    }
+}
+
+extension MoviesListState {
+    static func reduce(action: Action, state: MoviesListState) -> MoviesListState {
+        switch action {
+        case MoviesListAction.getMovies:
+            let movieAPI = MovieAPI()
+            let movieService = MovieService(movieAPI: movieAPI)
+            movieService.getMovies { (movies) in
+                mainStore.dispatch(MoviesListAction.saveMovies(movies: movies))
+            }
+            return MoviesListState(moviesList: .loading)
```

Comment:
Reducers can't instantiate services or API. Moreover, they shouldn't even know about anything except `State` and `Action`. In fact, `Store` and `Service` should be on the same level, while API can be reached only by `Service`. 
Let's move it into AppDelegate for now.